prow/plugins/updateconfig/updateconfig.go

```diff
@@ -117,19 +122,30 @@ func Update(fg FileGetter, kc corev1.ConfigMapInterface, name, namespace string,
 			return fmt.Errorf("get file err: %v", err)
 		}
 		logger.WithFields(logrus.Fields{"key": upd.Key, "filename": upd.Filename}).Debug("Populating key.")
-		value := string(content)
+		value := content
 		if upd.GZIP {
 			buff := bytes.NewBuffer([]byte{})
 			// TODO: this error is wildly unlikely for anything that
 			// would actually fit in a configmap, we could just as well return
 			// the error instead of falling back to the raw content
-			if _, err := gzip.NewWriter(buff).Write(content); err != nil {
+			z := gzip.NewWriter(buff)
+			if _, err := z.Write(content); err != nil {
 				logger.WithError(err).Error("failed to gzip content, falling back to raw")
 			} else {
-				value = buff.String()
+				if err := z.Close(); err != nil {
```

Comment:
Does a successful `Write()` always `Close()`? Can we defer the close?

Reply:
Closing isn't actually necessary in the traditional sense here - there's nothing OS-level being held open. We just need to actually get the complete output from `Write()`, which is buffered because... reasons? Not really sure, to be honest.

I think I could've used `z.Flush()` here to the same effect, but the docs vaguely imply you should be using `Close()` instead.

We can't _only_ defer the close, because we need it to be closed (really, flushed) before we call `z.Bytes()`, which would necessarily happen beforehand if we used `defer` (unless I added an anonymous function).